### PR TITLE
DBZ-5811 Implement support for skipping LSN commit on postgres source based on config

### PR DESCRIPTION
**Pull request by subodh1810**
Issue : https://issues.redhat.com/browse/DBZ-5811

### CODE REVIEW

**Comment by subodh1810** [comment]
@jpechane am not able to request a review. Can you help me with it?

**Comment by gunnarmorling** [comment]
Hey @jpechane, I got pointed to this PR; could you take a look?

Personally, I'm not quite convinced of this to be a good idea. I think the right philosophy for using PG replication slots is to consume them without much lag and relay that data elsewhere, preventing from WAL segments to be retained unnecessarily long. The risk for WAL to pile up is real, and odds are this falls back on Debezium also if it's actually due to bad user config/procedure.

To address the use case in the Jira issue, change events should be reprocessed from a durable log like Kafka (with topic retention set long enough to allow for any reprocessing needs of downstream systems).

**Comment by jpechane** [comment]
@gunnarmorling Hi, I was thinking about this quite hard and I really believe the feature request has its merit.
Look at it from other connector PoV. For other connectors the removeal of sourcing data is not driven by connector but by an external process - like script, admin work etc. Postgres connector is an exception. So I can really imagine that there are cases when it makes sense to have it to behave in the same manner as the other connectors. It is definitely plus if the connector is "self-cleaning" and correct default yet it might not be mmandatory.

**Comment by jpechane** [comment]
@subodh1810 Could you please take a look at the comment in Jira? It is correctly mentioned tha if you use Debezium Engine thn you can control offset commits yourself. Would that be a viable option?

**Comment by subodh1810** [comment]
@jpechane I posted a comment in Jira as well.

**Comment by subodh1810** [comment]
@jpechane is there anything additional required from my end for this to be reviewed?

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java [hunk 1]
```diff
@@ -52,6 +52,10 @@ public Class<? extends Task> taskClass() {
 
     @Override
     public void start(Map<String, String> props) {
+        if (props.getOrDefault(PostgresConnectorConfig.SHOULD_FLUSH_LSN_IN_SOURCE_DB.name(), "true").equalsIgnoreCase("false")) {
```
Comment (jpechane) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:55]: Can you please do this check in `POstgresConnectorConfig` in a dedicated validation method attached directly to `SHOULD_FLUSH_LSN_IN_SOURCE_DB` that would just emit warning and not fail?

**Review by jpechane — CHANGES_REQUESTED** [review]
@subodh1810 Looks good thanks! I've left few comments - no major change needed.

Could you please squash all commits into a single one and remove the merge commit? We prefer to have a linear history. Also could you please add the new option into the documentation?

Thanks

**Comment by rk3rn3r** [comment]
I am with Gunnar here. I also added a comment to the Jira issue.
This feature feels really like a custom functionality for a very, very specific use-case. I can imagine different ways of achieving the same without adding this functionality to DBZ (like when there's control over the source dbs replication slots, which could be reset in case of failure) or using other data integration solutions that better match such use-case. wdyt? Maybe the discussion should continue in the Jira.

**Comment by subodh1810** [comment]
@jpechane I have addressed your review comments, please take a look again

**Comment by subodh1810** [comment]
@jpechane the CI step `2022-12-05T12:13:34.1079986Z [INFO] Debezium Connector for PostgreSQL .................. FAILURE [19:43 min]
` seems to be failing with the following error 

```
2022-12-05T12:13:33.5284374Z [ERROR] Tests run: 89, Failures: 1, Errors: 2, Skipped: 0, Time elapsed: 240.802 s <<< FAILURE! - in io.debezium.connector.postgresql.RecordsStreamProducerIT
2022-12-05T12:13:33.5285431Z [ERROR] io.debezium.connector.postgresql.RecordsStreamProducerIT.shouldReceiveChangesForReplicaIdentityFullTableWithToastedValueTableFromStreamingFullDiff  Time elapsed: 3.127 s  <<< FAILURE!
2022-12-05T12:13:33.5286321Z org.junit.ComparisonFailure: expected:<null> but was:<Struct{pk=1}>
2022-12-05T12:13:33.5286792Z 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2022-12-05T12:13:33.5287399Z 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
2022-12-05T12:13:33.5288062Z 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2022-12-05T12:13:33.5288854Z 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
2022-12-05T12:13:33.5289288Z 	at io.debezium.data.VerifyRecord.isValidInsert(VerifyRecord.java:134)
2022-12-05T12:13:33.5289693Z 	at io.debezium.data.VerifyRecord.isValidInsert(VerifyRecord.java:237)
2022-12-05T12:13:33.5290260Z 	at io.debezium.connector.postgresql.RecordsStreamProducerIT.assertRecordInserted(RecordsStreamProducerIT.java:2984)
2022-12-05T12:13:33.5290956Z 	at io.debezium.connector.postgresql.RecordsStreamProducerIT.assertRecordInserted(RecordsStreamProducerIT.java:3014)
2022-12-05T12:13:33.5291625Z 	at io.debezium.connector.postgresql.RecordsStreamProducerIT.assertInsert(RecordsStreamProducerIT.java:2947)
2022-12-05T12:13:33.5292237Z 	at io.debezium.connector.postgresql.RecordsStreamProducerIT.assertInsert(RecordsStreamProducerIT.java:2937)
2022-12-05T12:13:33.5293092Z 	at io.debezium.connector.postgresql.RecordsStreamProducerIT.testReceiveChangesForReplicaIdentityFullTableWithToastedValue(RecordsStreamProducerIT.java:2610)
2022-12-05T12:13:33.5294365Z 	at io.debezium.connector.postgresql.RecordsStreamProducerIT.shouldReceiveChangesForReplicaIdentityFullTableWithToastedValueTableFromStreamingFullDiff(RecordsStreamProducerIT.java:1672)
2022-12-05T12:13:33.5295507Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-12-05T12:13:33.5296019Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-12-05T12:13:33.5296590Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-12-05T12:13:33.5297063Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-12-05T12:13:33.5297484Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2022-12-05T12:13:33.5297957Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2022-12-05T12:13:33.5298473Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2022-12-05T12:13:33.5299141Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2022-12-05T12:13:33.5299631Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2022-12-05T12:13:33.5300116Z 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
2022-12-05T12:13:33.5300524Z 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
2022-12-05T12:13:33.5300882Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-12-05T12:13:33.5301306Z 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
2022-12-05T12:13:33.5301806Z 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
2022-12-05T12:13:33.5302243Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
2022-12-05T12:13:33.5302722Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
2022-12-05T12:13:33.5303140Z 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2022-12-05T12:13:33.5303500Z 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2022-12-05T12:13:33.5303887Z 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2022-12-05T12:13:33.5304279Z 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2022-12-05T12:13:33.5304654Z 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2022-12-05T12:13:33.5305022Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-12-05T12:13:33.5305387Z 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2022-12-05T12:13:33.5305805Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:377)
2022-12-05T12:13:33.5321869Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:284)
2022-12-05T12:13:33.5322438Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:248)
2022-12-05T12:13:33.5322959Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:167)
2022-12-05T12:13:33.5323469Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
2022-12-05T12:13:33.5323980Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
2022-12-05T12:13:33.5324436Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
2022-12-05T12:13:33.5324877Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
2022-12-05T12:13:33.5325119Z 
2022-12-05T12:13:33.5325627Z [ERROR] io.debezium.connector.postgresql.RecordsStreamProducerIT.shouldReceiveChangesForDeletesDependingOnReplicaIdentity  Time elapsed: 3.017 s  <<< ERROR!
2022-12-05T12:13:33.5326268Z java.lang.RuntimeException: 
2022-12-05T12:13:33.5326696Z org.postgresql.util.PSQLException: ERROR: cannot delete from table "test_table" because it does not have a replica identity and publishes deletes
2022-12-05T12:13:33.5327166Z   Hint: To enable deleting from the table, set REPLICA IDENTITY using ALTER TABLE.
2022-12-05T12:13:33.5327584Z 	at io.debezium.connector.postgresql.TestHelper.execute(TestHelper.java:192)
2022-12-05T12:13:33.5328336Z 	at io.debezium.connector.postgresql.RecordsStreamProducerIT.executeAndWait(RecordsStreamProducerIT.java:3018)
2022-12-05T12:13:33.5329733Z 	at io.debezium.connector.postgresql.RecordsStreamProducerIT.shouldReceiveChangesForDeletesDependingOnReplicaIdentity(RecordsStreamProducerIT.java:1045)
2022-12-05T12:13:33.5330486Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-12-05T12:13:33.5331002Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-12-05T12:13:33.5331574Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-12-05T12:13:33.5332040Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-12-05T12:13:33.5332613Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2022-12-05T12:13:33.5333098Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2022-12-05T12:13:33.5333586Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2022-12-05T12:13:33.5334097Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2022-12-05T12:13:33.5334585Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2022-12-05T12:13:33.5335056Z 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
2022-12-05T12:13:33.5335535Z 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
2022-12-05T12:13:33.5335905Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-12-05T12:13:33.5336319Z 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
2022-12-05T12:13:33.5336737Z 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
2022-12-05T12:13:33.5337170Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
2022-12-05T12:13:33.5337633Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
2022-12-05T12:13:33.5338046Z 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2022-12-05T12:13:33.5338414Z 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2022-12-05T12:13:33.5338852Z 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2022-12-05T12:13:33.5339236Z 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2022-12-05T12:13:33.5339609Z 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2022-12-05T12:13:33.5339977Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-12-05T12:13:33.5340334Z 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2022-12-05T12:13:33.5340764Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:377)
2022-12-05T12:13:33.5341286Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:284)
2022-12-05T12:13:33.5341812Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:248)
2022-12-05T12:13:33.5342305Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:167)
2022-12-05T12:13:33.5342809Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
2022-12-05T12:13:33.5343313Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
2022-12-05T12:13:33.5343766Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
2022-12-05T12:13:33.5344216Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
2022-12-05T12:13:33.5344731Z Caused by: org.postgresql.util.PSQLException: ERROR: cannot delete from table "test_table" because it does not have a replica identity and publishes deletes
2022-12-05T12:13:33.5345215Z   Hint: To enable deleting from the table, set REPLICA IDENTITY using ALTER TABLE.
2022-12-05T12:13:33.5345677Z 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2676)
2022-12-05T12:13:33.5346200Z 	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2366)
2022-12-05T12:13:33.5346693Z 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:356)
2022-12-05T12:13:33.5347138Z 	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:496)
2022-12-05T12:13:33.5347545Z 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:413)
2022-12-05T12:13:33.5347951Z 	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:333)
2022-12-05T12:13:33.5348385Z 	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:319)
2022-12-05T12:13:33.5348808Z 	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:295)
2022-12-05T12:13:33.5349253Z 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:290)
2022-12-05T12:13:33.5349694Z 	at io.debezium.jdbc.JdbcConnection.executeWithoutCommitting(JdbcConnection.java:1440)
2022-12-05T12:13:33.5350167Z 	at io.debezium.connector.postgresql.TestHelper.execute(TestHelper.java:179)
2022-12-05T12:13:33.5350465Z 	... 33 more
2022-12-05T12:13:33.5350587Z 
2022-12-05T12:13:33.5350957Z [ERROR] io.debezium.connector.postgresql.RecordsStreamProducerIT.shouldReceiveChangesForUpdates  Time elapsed: 3.564 s  <<< ERROR!
2022-12-05T12:13:33.5351445Z java.lang.RuntimeException: 
2022-12-05T12:13:33.5351865Z org.postgresql.util.PSQLException: ERROR: cannot update table "test_table" because it does not have a replica identity and publishes updates
2022-12-05T12:13:33.5352360Z   Hint: To enable updating the table, set REPLICA IDENTITY using ALTER TABLE.
2022-12-05T12:13:33.5352768Z 	at io.debezium.connector.postgresql.TestHelper.execute(TestHelper.java:192)
2022-12-05T12:13:33.5353375Z 	at io.debezium.connector.postgresql.RecordsStreamProducerIT.executeAndWaitForNoRecords(RecordsStreamProducerIT.java:3023)
2022-12-05T12:13:33.5354132Z 	at io.debezium.connector.postgresql.RecordsStreamProducerIT.shouldReceiveChangesForUpdates(RecordsStreamProducerIT.java:672)
2022-12-05T12:13:33.5354750Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-12-05T12:13:33.5355256Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-12-05T12:13:33.5355814Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-12-05T12:13:33.5356276Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-12-05T12:13:33.5356682Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2022-12-05T12:13:33.5357159Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2022-12-05T12:13:33.5357645Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2022-12-05T12:13:33.5358139Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2022-12-05T12:13:33.5358624Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2022-12-05T12:13:33.5359086Z 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
2022-12-05T12:13:33.5359500Z 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
2022-12-05T12:13:33.5359866Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-12-05T12:13:33.5360280Z 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
2022-12-05T12:13:33.5360704Z 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
2022-12-05T12:13:33.5361132Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
2022-12-05T12:13:33.5361604Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
2022-12-05T12:13:33.5362016Z 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2022-12-05T12:13:33.5362376Z 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2022-12-05T12:13:33.5362759Z 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2022-12-05T12:13:33.5363148Z 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2022-12-05T12:13:33.5363518Z 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2022-12-05T12:13:33.5363883Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-12-05T12:13:33.5364242Z 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2022-12-05T12:13:33.5364666Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:377)
2022-12-05T12:13:33.5365171Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:284)
2022-12-05T12:13:33.5365699Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:248)
2022-12-05T12:13:33.5366245Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:167)
2022-12-05T12:13:33.5366740Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
2022-12-05T12:13:33.5367240Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
2022-12-05T12:13:33.5367690Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
2022-12-05T12:13:33.5368527Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
2022-12-05T12:13:33.5369067Z Caused by: org.postgresql.util.PSQLException: ERROR: cannot update table "test_table" because it does not have a replica identity and publishes updates
2022-12-05T12:13:33.5372169Z   Hint: To enable updating the table, set REPLICA IDENTITY using ALTER TABLE.
2022-12-05T12:13:33.5373089Z 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2676)
2022-12-05T12:13:33.5373627Z 	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2366)
2022-12-05T12:13:33.5374125Z 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:356)
2022-12-05T12:13:33.5374579Z 	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:496)
2022-12-05T12:13:33.5374999Z 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:413)
2022-12-05T12:13:33.5375407Z 	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:333)
2022-12-05T12:13:33.5375847Z 	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:319)
2022-12-05T12:13:33.5376269Z 	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:295)
2022-12-05T12:13:33.5376670Z 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:290)
2022-12-05T12:13:33.5377124Z 	at io.debezium.jdbc.JdbcConnection.executeWithoutCommitting(JdbcConnection.java:1440)
2022-12-05T12:13:33.5377596Z 	at io.debezium.connector.postgresql.TestHelper.execute(TestHelper.java:179)
2022-12-05T12:13:33.5377899Z 	... 33 more
```

This seems un-related to my changes (please correct me if am wrong). Can you help me out here please.

**Comment by roldanbob** [comment]
@jpechane Do we intend that this change is for both the community and product versions?  Prior to merging any documentation changes that are not conditionalized for community-only use, _please_ request my review. I'll do my best to respond in a timely fashion. The property description here requires several style and consistency edits, and it's always  easier to do implement those before a change is merged.

**Pull request merged**